src/React.Core/ReactIdGenerator.cs

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright (c) 2016-Present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant 
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+using System;
+using System.Threading;
+
+namespace React
+{
+	/// <summary>
+	/// Extension methods relating to GUIDs.
+	/// </summary>
+	public static class ReactIdGenerator
+	{
+		private static readonly string _encode32Chars = "0123456789ABCDEFGHIJKLMNOPQRSTUV";
+
+		private static long _random = DateTime.UtcNow.Ticks;
+
+		private static readonly char[] reactPrefix = "react_".ToCharArray();
+
+		[ThreadStatic]
+		private static char[] _chars;
+
+		/// <summary>
+		/// Returns a short react identifier starts with "react_".
+		/// </summary>
+		/// <returns></returns>
+		public static string Generate()
+		{
+			var chars = _chars;
+			if (chars == null)
+			{
+				_chars = chars = new char[19];
+				Array.Copy(reactPrefix, 0, chars, 0, reactPrefix.Length);
+			}
+
+			var id = Interlocked.Increment(ref _random);
+
+			//reactPrefix.Length == 6
+			chars[6] = _encode32Chars[(int)(id >> 60) & 31];
```

Comment:
There's a lot of magic numbers in this code. It'd be good to pull these out as constants and add comments explaining what they're doing.